packages/electron-updater/src/MacUpdater.ts

```diff
@@ -94,16 +94,15 @@ export class MacUpdater extends AppUpdater {
 
     const provider = downloadUpdateOptions.updateInfoAndProvider.provider
     const CURRENT_MAC_APP_ZIP_FILE_NAME = "update.zip"
-    let cachedUpdateFile: string = ""
 
     return this.executeDownload({
       fileExtension: "zip",
       fileInfo: zipFileInfo,
       downloadUpdateOptions,
       task: async (destinationFile, downloadOptions) => {
-        cachedUpdateFile = path.join(this.downloadedUpdateHelper!.cacheDir, CURRENT_MAC_APP_ZIP_FILE_NAME)
+        const cachedUpdateFilePath = path.join(this.downloadedUpdateHelper!.cacheDir, CURRENT_MAC_APP_ZIP_FILE_NAME)
         const canDifferentialDownload = () => {
-          if (!pathExistsSync(cachedUpdateFile)) {
+          if (!pathExistsSync(cachedUpdateFilePath)) {
```

Comment:
This looks good, but I'm curious as to why `cacheDir` is an empty string. Won't this `!pathExistsSync(...)` always return false and fallback to full download?

Reply:
https://github.com/electron-userland/electron-builder/blob/3b03f27245458a0b595b7a65c8634f184d32f2fa/packages/electron-updater/src/AppUpdater.ts#L698-L702
The problem likely lies here. If cachedUpdateFile already exists, the task function won't run; instead, the done function will be executed directly.